From 15845d21cdc4be3cafa66ee5841a10531802b33d Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Sun, 25 Nov 2018 18:38:26 +0200
Subject: [PATCH 0808/1051] arm64: dts: a37xx: re-arrange SDHCI controllers
 order

Sync the controllers order with uboot settings:
sdhci-0 is @ 0xd000, sdhci-1 is @ 0xd800.
Add pin controller settings for SD and eMMC

Change-Id: Ie4e0fa1167a63a7e4eb31a1c12738bb5bd7d734b
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61291
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts   | 16 ++++++++--------
 .../boot/dts/marvell/armada-3720-espressobin.dts | 13 ++++++++-----
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi     | 14 ++++++++++++--
 3 files changed, 28 insertions(+), 15 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 5dfa38e0ae9b..8aabaada2f5e 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -59,9 +59,9 @@
 		vcc-supply = <&exp_usb2_vbus>;
 	};
 
-	vcc_sd_reg1: regulator {
+	vcc_sd_reg0: regulator {
 		compatible = "regulator-gpio";
-		regulator-name = "vcc_sd1";
+		regulator-name = "vcc_sd0";
 		regulator-min-microvolt = <1800000>;
 		regulator-max-microvolt = <3300000>;
 		regulator-boot-on;
@@ -73,9 +73,9 @@
 		enable-active-high;
 	};
 
-	vcc_sd_reg2: regulator-vmcc {
+	vcc_sd_reg1: regulator-vmcc {
 		compatible = "regulator-fixed";
-		regulator-name = "vcc_sd2";
+		regulator-name = "vcc_sd1";
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
 		regulator-boot-on;
@@ -145,7 +145,7 @@
 	phys = <&a3700_comphy2 0>;
 };
 
-&sdhci0 {
+&sdhci1 {
 	non-removable;
 	bus-width = <8>;
 	mmc-ddr-1_8v;
@@ -155,13 +155,13 @@
 };
 
 /* SD slot module on CON14(V2.0)/CON15(V1.4) */
-&sdhci1 {
+&sdhci0 {
 	wp-inverted;
 	cd-gpios = <&gpiosb 2 GPIO_ACTIVE_LOW>;
 	bus-width = <4>;
 	marvell,pad-type = "sd";
-	vqmmc-supply = <&vcc_sd_reg1>;
-	vmmc-supply = <&vcc_sd_reg2>;
+	vqmmc-supply = <&vcc_sd_reg0>;
+	vmmc-supply = <&vcc_sd_reg1>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-espressobin.dts b/arch/arm64/boot/dts/marvell/armada-3720-espressobin.dts
index a5e096a87ad9..488ff5a19266 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-espressobin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-espressobin.dts
@@ -17,7 +17,8 @@
 
 / {
 	model = "Globalscale Marvell ESPRESSOBin Board";
-	compatible = "globalscale,espressobin", "marvell,armada3720", "marvell,armada3710";
+	compatible = "globalscale,espressobin",
+		     "marvell,armada3720", "marvell,armada3710";
 
 	chosen {
 		stdout-path = "serial0:115200n8";
@@ -28,9 +29,9 @@
 		reg = <0x00000000 0x00000000 0x00000000 0x20000000>;
 	};
 
-	vcc_sd_reg1: regulator {
+	vcc_sd_reg0: regulator {
 		compatible = "regulator-gpio";
-		regulator-name = "vcc_sd1";
+		regulator-name = "vcc_sd0";
 		regulator-min-microvolt = <1800000>;
 		regulator-max-microvolt = <3300000>;
 		regulator-boot-on;
@@ -58,12 +59,14 @@
 };
 
 /* J1 */
-&sdhci1 {
+&sdhci0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&sdio_pins>;
 	wp-inverted;
 	bus-width = <4>;
 	cd-gpios = <&gpionb 3 GPIO_ACTIVE_LOW>;
 	marvell,pad-type = "sd";
-	vqmmc-supply = <&vcc_sd_reg1>;
+	vqmmc-supply = <&vcc_sd_reg0>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index a03efc7b787d..e1b4d50d95b6 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -261,6 +261,11 @@
 					groups = "uart2";
 					function = "uart";
 				};
+
+				mmc_pins: mmc-pins {
+					groups = "emmc_nb";
+					function = "emmc";
+				};
 			};
 
 			nb_pm: syscon@14000 {
@@ -293,6 +298,11 @@
 					function = "mii";
 				};
 
+				sdio_pins: sdio-pins {
+					groups = "sdio_sb";
+					function = "sdio";
+				};
+
 			};
 
 			eth0: ethernet@30000 {
@@ -370,7 +380,7 @@
 				cell-index = <0>;
 			};
 
-			sdhci1: sdhci@d0000 {
+			sdhci0: sdhci@d0000 {
 				compatible = "marvell,armada-3700-sdhci",
 					     "marvell,sdhci-xenon";
 				reg = <0xd0000 0x300>,
@@ -381,7 +391,7 @@
 				status = "disabled";
 			};
 
-			sdhci0: sdhci@d8000 {
+			sdhci1: sdhci@d8000 {
 				compatible = "marvell,armada-3700-sdhci",
 					     "marvell,sdhci-xenon";
 				reg = <0xd8000 0x300>,
-- 
2.17.1

